olp-cpp-sdk-dataservice-write/src/CatalogSettings.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "CatalogSettings.h"
+
+#include <olp/core/cache/CacheSettings.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <boost/format.hpp>
+#include "ApiClientLookup.h"
+#include "generated/ConfigApi.h"
+
+// clang-format off
+#include <generated/serializer/CatalogSerializer.h>
+#include <generated/serializer/JsonSerializer.h>
+#include <generated/parser/CatalogParser.h>
+#include <olp/core/generated/parser/JsonParser.h>
+// clang-format on
+
+namespace olp {
+namespace dataservice {
+namespace write {
+
+CatalogSettings::CatalogSettings(const client::HRN catalog,
+                                 client::OlpClientSettings settings)
+    : catalog_(std::move(catalog)),
+      cache_(settings.cache),
+      settings_(std::move(settings)) {
+  if (!settings.cache) {
+    settings.cache = client::OlpClientSettingsFactory::CreateDefaultCache({});
+    cache_ = settings.cache;
```

Comment:
```suggestion
      settings_(std::move(settings)) {
  if (!settings_.cache) {
    settings_.cache = client::OlpClientSettingsFactory::CreateDefaultCache({});
    cache_ = settings_.cache;
```
settings already moved here.